hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -322,7 +323,9 @@ object HoodieSparkUtils extends SparkAdapterSupport {
       val name2Fields = tableAvroSchema.getFields.asScala.map(f => f.name() -> f).toMap
       // Here have to create a new Schema.Field object
       // to prevent throwing exceptions like "org.apache.avro.AvroRuntimeException: Field already used".
-      val requiredFields = requiredColumns.map(c => name2Fields(c))
+      // For a nested field, we include the root-level field
+      val requiredFields = requiredColumns.map(c => HoodieAvroUtils.getRootLevelFieldName(c))
+        .distinct.map(c => name2Fields(c))
```

Comment:
this code path affects both MOR and COW, right?  is it for spark later to drill down to specific nested cols?

Reply:
just MOR.

Reply:
we discussed the `distinct` looks problematic.